language-python.md

```diff
@@ -0,0 +1,106 @@
+* [Supported Versions](#supported-versions)
+* [Dependency Caching](#dependency-caching)
+* [Environment Variables](#environment-variables)
+* [C-Extensions & Dependencies](#c-extensions-dependendices)
+
+## Supported Versions
+
+Python 2.7 is the default version. This can be switched to 3.7 with
+`sem-version`. Here's an example:
+
+<pre><code class="language-yaml">blocks:
+  - name: Tests
+    task:
+      prologue:
+        commands:
+          - sem-version python 3.7
+      jobs:
+        - name: Tests
+          commands:
+            - python --version
+</code></pre>
+
+## Dependency Caching
+
+You can use Semaphore's `cache` command to store and load `pipenv`'s
+virtualenv. This requires setting the `PIPENV_VENV_IN_PROJECT`
+environment variable. You'll need one block to warm the cache, then
+use the cache in subsequent blocks.
+
+<pre><code class="language-yaml">version: "v1.0"
+version: "v1.0"
+name: Python Example
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: Warm caches
+    task:
+      env_vars:
+        - name: PIPENV_VENV_IN_PROJECT
+          value: "true"
+      prologue:
+        commands:
+          - sudo pip install pipenv
```

Comment:
I suggest `pipenv` be installed as part of standard ubuntu image. It's askin to hav having ruby without bundler.